plugins/fastem_script_runner.py

```diff
@@ -0,0 +1,627 @@
+        version_box = wx.StaticBox(panel, label="Select Version:")
+        version_box_sizer = wx.StaticBoxSizer(version_box, wx.VERTICAL)
+        self.version_radio_box = wx.RadioBox(
+            panel,
+            choices=list(self.version_base_path.keys()),
+            majorDimension=1,  # 1 column
+            style=wx.RA_SPECIFY_COLS,
+        )
+        self.version_radio_box.SetSelection(0)  # Default to Official Release
+        version_box_sizer.Add(self.version_radio_box, 0, wx.EXPAND | wx.ALL, 5)
+        main_sizer.Add(version_box_sizer, 0, wx.EXPAND | wx.ALL, 10)
+
+        # Buttons
+        btn_sizer = wx.StdDialogButtonSizer()
+        ok_btn = wx.Button(panel, wx.ID_OK)
+        ok_btn.SetDefault()
+        cancel_btn = wx.Button(panel, wx.ID_CANCEL)
+        btn_sizer.AddButton(ok_btn)
+        btn_sizer.AddButton(cancel_btn)
+        btn_sizer.Realize()
+        main_sizer.Add(btn_sizer, 0, wx.ALIGN_CENTER | wx.ALL, 10)
+
+        panel.SetSizer(main_sizer)
+        main_sizer.Fit(self)  # Fit dialog to contents
+        self.SetMinSize(self.GetSize())  # Prevent shrinking too much
+        self.CentreOnParent()
+
+        ok_btn.Bind(wx.EVT_BUTTON, self.on_ok)
+
+    def on_ok(self, event):
+        """Handle the OK button click event."""
+        script_idx = self.script_choice.GetSelection()
+        if script_idx == wx.NOT_FOUND:
+            wx.MessageBox(
+                "Please select a script.",
+                "Selection Error",
+                wx.OK | wx.ICON_ERROR,
+                self,
+            )
+            return
+
+        _, script_name = self.available_scripts[script_idx]
+        version_idx = self.version_radio_box.GetSelection()
+        version_str = self.version_radio_box.GetString(version_idx)
+
+        try:
+            base_path, chosen_path = find_script_path(version_str, script_name, self.version_base_path)
+        except (ValueError, FileNotFoundError) as e:
+            wx.MessageBox(
+                str(e),
+                "Error Finding Script",
+                wx.OK | wx.ICON_ERROR,
+                self,
+            )
+            return
+
+        try:
+            if version_str == "Development":
+                # Normalize the base_path to avoid issues with trailing slashes, etc.
+                normalized_base_path = os.path.abspath(os.path.normpath(base_path))
+
+                # Get the current PYTHONPATH from the environment Odemis is running in
+                current_pythonpath_str = os.environ.get("PYTHONPATH", "")
+                current_pythonpath_list = [
+                    os.path.abspath(os.path.normpath(p))
+                    for p in current_pythonpath_str.split(os.pathsep)
+                    if p
+                ]
+
+                # Prepend the new base_path
+                new_path_list = [normalized_base_path] + current_pythonpath_list
+                self.script_env["PYTHONPATH"] = os.pathsep.join(new_path_list)
```

Comment:
You don't need to convert from string to list, and back to string.
```suggestion
                current_pythonpath = os.environ.get("PYTHONPATH", "")
                # Prepend the new base_path
                self.script_env["PYTHONPATH"] = normalized_base_path + os.pathsep + current_pythonpath
```